### PR TITLE
Fix keyboard disappearing while switching between text fields

### PR DESCRIPTION
**Pull request by ASalavei**
https://github.com/JetBrains/compose-multiplatform/issues/3856

### CODE REVIEW

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitTextInputService.uikit.kt [hunk 1]
```diff
@@ -127,7 +130,14 @@ internal class UIKitTextInputService(
         currentImeOptions = null
         currentImeActionHandler = null
         hideSoftwareKeyboard()
-        textUIView?.removeFromSuperview()
+
+        textUIView?.input = null
+        textUIView?.let { view ->
+
+            dispatch_async(dispatch_get_main_queue()) {
```
Comment (ASalavei) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitTextInputService.uikit.kt]: @dima-avdeev-jb , do you have something in your mind how to get similar effect using coroutines?

Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitTextInputService.uikit.kt]: In coroutines we have `yield()` function

**Review by dima-avdeev-jb — APPROVED** [review]
Brilliant!

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitTextInputService.uikit.kt (hunk 1 above)
Comment (ASalavei) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitTextInputService.uikit.kt]: Done!

**Pull request merged**